packages/python/plotly/plotly/express/_core.py

```diff
@@ -1915,29 +1915,43 @@ def get_orderings(args, grouper, grouped):
     of tuples like [("value1", ""), ("value2", "")] where each tuple contains the name
     of a single dimension-group
     """
-
     orders = {} if "category_orders" not in args else args["category_orders"].copy()
+
+    # figure out orders and what the single group name would be if there were one
+    single_group_name = []
     for col in grouper:
-        if col != one_group:
+        if col == one_group:
+            single_group_name.append("")
+        else:
             uniques = list(args["data_frame"][col].unique())
```

Comment:
For the following snippet this line is executed twice for the exact same col ("variable");

```py
import plotly.express as px
px.line([1, 2, 3, 4])
```

I guess that the second computation could be avoided? (this is why I used `set(grouper)` to construct the order dictionary)

Reply:
We could memoize the results of this loop as a function of `col`, yeah

Reply:
(done)